department-of-veterans-affairs/caseflow
# Create Court Remand (CAVC) Appeal Stream on CavcRemandForm Completion

Resolves #14792 

### Description
Please explain the changes you made here.

### Acceptance Criteria
- [ ] (Nothing shows) ~If anything would show to users before CAVC Remands enter the system, lock that behind FeatureToggle: cavc_remand~
- [ ] New appeal stream is created when Litigation Support team chooses a JMR or JMPR and clicks submit on the court order form
- [ ] New appeal stream is linked to the existing docket number at the Board
- [ ] Create Root Case, Distribution Task and CavcTask
- [ ] Add instances of new appeal stream to seed data
- [ ] Verify the original and new appeal appears when searching for original Docket #

### Testing Plan
All via console for now - see reference code below
1. Create dispatched appeal and create a JMR CavcRemands
   - [x] No Errors
   - [x] There is now a CAVC appeal stream with the same issues
1. Create dispatched appeal and create a JMPR CavcRemands
   - [x] No Errors
   - [x] There is now a CAVC appeal stream with the subset of issues
1. Create a CavcRemand factory object, default
   - [x] No Errors
   - [x] There is a dispatched original appeal stream associated with the CavcRemand
   - [x] There is now a CAVC appeal stream
1. Create a CavcRemand factory object, pass in an Appeal
   - [x] No Errors
   - [x] The provided appeal stream associated with the CavcRemand
   - [x] There is now a CAVC appeal stream with request issues matching the provided Appeal's decision issues
1. Confirm the Search AC in the GUI
1. Confirm the db seeding works, and Disptached & CAVC cases are created

### Reference code!
Create a ready for CAVC appeal
```ruby
original_appeal = FactoryBot.create(:appeal, 
                           :dispatched, 
                           associated_judge: JudgeTeam.first.judge, 
                           associated_attorney: JudgeTeam.first.attorneys.first)
FactoryBot.create_list(:request_issue, 2,                                                                                  
            :rating,                                                                                            
            :with_rating_decision_issue,                                                                           
            decision_review: original_appeal,                                                                               
            veteran_participant_id: Veteran.first.participant_id,                                                        
            contested_issue_description: "issue_description",                                                              
            notes: "issue_notes")
```
Create JMR/JMPR CavcRemand
```ruby
params = {
  created_by: User.first,
  appeal: original_appeal,
  cavc_docket_number: 1234,
  represented_by_attorney: true,
  cavc_judge_full_name: Constants::CAVC_JUDGE_FULL_NAMES.first,
  cavc_decision_type: Constants::CAVC_DECISION_TYPES.keys.first,
  decision_date: 2.days.ago,
  judgement_date: 2.days.ago,
  mandate_date: 2.days.ago,
  instructions: "cavc instructions"}

jmr_params = params.merge({remand_subtype: Constants::CAVC_REMAND_SUBTYPES.keys.first,
                           decision_issue_ids: original_appeal.decision_issues.pluck(:id)})
jmr_cavc_remand = CavcRemand.create!(jmr_params)

jmpr_params = params.merge({remand_subtype: Constants::CAVC_REMAND_SUBTYPES.keys.second,
                           decision_issue_ids: [original_appeal.decision_issues.first.id]})
jmpr_cavc_remand = CavcRemand.create!(jmr_params)
```
Find CAVC/Court Remand Stream Appeals
```ruby
Appeal.court_remand.where(stream_docket_number: original_appeal.docket_number)
```

Create a CavcRemand factory:
```ruby
cavc_remand = FactoryBot.create(:cavc_remand)
```
